app/controllers/uploader/files_controller.rb

```diff
@@ -1,169 +1,169 @@
+    files = @items.select{ |item| !item.directory? }.sort_by { |item| item.name.capitalize }
+    @items = dirs + files
+    @items.each { |item| item.site = @cur_site }
+  end
+
+  def create_files
+    @files.each do |file|
+      next unless file.present?
+      path = ::File.join(@cur_site.path, @item.filename, file.original_filename)
+      item = @model.new(path: path, binary: file.read, site: @cur_site)
+
+      if !item.save
+        item.errors.each do |n, e|
+          @item.errors.add :base, "#{item.name} - #{@model.t(n)}#{e}"
+        end
+      end
+    end
+    location = "#{uploader_files_path}/#{@item.filename}"
+    render_create @item.errors.empty?, location: location, render: { file: :new_files }
+  end
+
+  def create_directory
+    path = "#{@item.path}/#{@directory}"
+    item = @model.new(path: path, is_dir: true, site: @cur_site)
+
+    if !item.save
+      item.errors.each do |n, e|
+        @item.errors.add :path, e
+      end
+    end
+    location = "#{uploader_files_path}/#{@item.filename}"
+    render_create @item.errors.empty?, location: location, render: { file: :new_directory }
+  end
+
+  def set_params(*keys)
+    keys.each { |key| instance_variable_set("@#{key}", params[:item].try(:[], key)) }
+  rescue => e
+    Rails.logger.debug("#{e.class} (#{e.message}):\n  #{e.backtrace.join("\n  ")}")
+    raise "400"
+  end
+
+  public
+  def file
+    action = params[:do] || "index"
+    raise "404" unless @item
+    raise "404" unless %w(index new_directory new_files show edit delete).index(action)
+    send action
+  end
+
+  def index
+    raise "404" unless @item.directory?
+    raise "403" unless @cur_node.allowed?(:read, @cur_user, site: @cur_site)
+    set_items(@item.path)
+    render :index
+  end
+
+  def edit
+    raise "403" unless @cur_node.allowed?(:edit, @cur_user, site: @cur_site)
+    render :edit
+  end
+
+  def show
+    raise "403" unless @cur_node.allowed?(:read, @cur_user, site: @cur_site)
+    render :show
+  end
+
+  def delete
+    raise "403" unless @cur_node.allowed?(:delete, @cur_user, site: @cur_site)
+    render :delete
+  end
+
+  def new_files
+    raise "403" unless @cur_node.allowed?(:edit, @cur_user, site: @cur_site)
+    render :new_files
+  end
+
+  def new_directory
+    raise "403" unless @cur_node.allowed?(:edit, @cur_user, site: @cur_site)
+    render :new_directory
+  end
+
+  def create
+    set_params(:directory, :files)
+    if @directory
+      create_directory
+    elsif @files
+      create_files
+    else
+      raise "400"
+    end
+  end
+
+  def update
+    set_params(:filename, :files, :text, :file)
+    raise "403" unless @cur_node.allowed?(:edit, @cur_user, site: @cur_site)
+    raise "400" unless @filename
+
+    if !@item.directory?
+      @text ? (@item.text = @text) : @item.read
+    end
+    @item.filename = @filename if @filename && @filename =~ /^#{@cur_node.filename}/
+    @item.site = @cur_site
```

Comment:
__[brakeman]__
* Model attribute used in regex [:ref](http://brakemanscanner.org/docs/warning_types/denial_of_service/)

<!-- Comment by [SideCI](https://sideci.com) -->